File: src/main/java/com/shop/entity/Owner.java
package com.shop.entity;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import jakarta.persistence.*;
import lombok.Data;
@Data
@Entity
@Table(name = "Owner")
public class Owner {
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	private Long id;
	@Column(name="email")
	private String email;
	@Column(name="pwd")
	private String pwd;
	@Column(name="dob")
	private Date dob = new Date();
	@Column(name="firstName")
	private String firstName;
	@Column(name="lastName")
	private String lastName;
	@Column(name="gender")
	private String gender;
	@Column(name = "businessName")
	private String businessName = "ShopFlow";
	@ManyToMany(targetEntity=Customer.class)
	private List<Customer> customers = new ArrayList<>();
}

File: src/app/owner.ts
export class Owner {
    id?: number;
    email?: string;
    pwd?: string;
    dob: string; 
    firstName?: string;
    lastName?: string;
    gender?: string;

    constructor() {
        this.dob = this.formatDate(new Date());
    }

    private formatDate(date: Date): string {
        const year = date.getFullYear();
        const month = (date.getMonth() + 1).toString().padStart(2, '0');
        const day = date.getDate().toString().padStart(2, '0');
        return `${year}-${month}-${day}`;
    }
}
